server/index.js

```diff
@@ -0,0 +1,452 @@
+// that the bank website should redirect to. You will need to whitelist
+// this redirect URI for your client ID through the Plaid developer dashboard
+// at https://dashboard.plaid.com/team/api.
+var PLAID_OAUTH_REDIRECT_URI = envvar.string('PLAID_OAUTH_REDIRECT_URI', '');
+// Set PLAID_OAUTH_NONCE to a unique identifier such as a UUID for each Link
+// session. The nonce will be used to re-open Link upon completion of the OAuth
+// redirect. The nonce must be at least 16 characters long.
+var PLAID_OAUTH_NONCE = envvar.string('PLAID_OAUTH_NONCE', '');
+
+// We store the access_token in memory - in production, store it in a secure
+// persistent data store
+var ACCESS_TOKEN = null;
+var PUBLIC_TOKEN = null;
+var ITEM_ADD_TOKEN = null;
+
+var ITEM_ID = null;
+// The payment_token is only relevant for the UK Payment Initiation product.
+// We store the payment_token in memory - in production, store it in a secure
+// persistent data store
+var PAYMENT_TOKEN = null;
+var PAYMENT_ID = null;
+
+// Initialize the Plaid client
+// Find your API keys in the Dashboard (https://dashboard.plaid.com/account/keys)
+var client = new plaid.Client(
+  PLAID_CLIENT_ID,
+  PLAID_SECRET,
+  PLAID_PUBLIC_KEY,
+  plaid.environments[PLAID_ENV],
+  {version: '2019-05-29', clientApp: 'Plaid Quickstart'}
+);
+
+var app = express();
+app.use(express.static('public'));
+app.set('view engine', 'ejs');
+app.use(bodyParser.urlencoded({
+  extended: false
+}));
+app.use(bodyParser.json());
+
+app.get('/', function(request, response, next) {
+  response.render('index.ejs', {
+    PLAID_PUBLIC_KEY: PLAID_PUBLIC_KEY,
+    PLAID_ENV: PLAID_ENV,
+    PLAID_PRODUCTS: PLAID_PRODUCTS,
+    PLAID_COUNTRY_CODES: PLAID_COUNTRY_CODES,
+    PLAID_OAUTH_REDIRECT_URI: PLAID_OAUTH_REDIRECT_URI,
+    PLAID_OAUTH_NONCE: PLAID_OAUTH_NONCE,
+    ITEM_ID: ITEM_ID,
+    ACCESS_TOKEN: ACCESS_TOKEN,
+  });
+});
+
+// This is an endpoint defined for the OAuth flow to redirect to.
+app.get('/oauth-response.html', function(request, response, next) {
+  response.render('oauth-response.ejs', {
+    PLAID_PUBLIC_KEY: PLAID_PUBLIC_KEY,
+    PLAID_ENV: PLAID_ENV,
+    PLAID_PRODUCTS: PLAID_PRODUCTS,
+    PLAID_COUNTRY_CODES: PLAID_COUNTRY_CODES,
+    PLAID_OAUTH_NONCE: PLAID_OAUTH_NONCE,
+  });
+});
+
+
+// Add Item Token Flow - Create new items by using an item_add_token
+app.post('/get_add_token', function(request, response, next) {
+  client.createItemAddToken({
+    user: {
+      client_user_id: PLAID_CLIENT_ID,
+    },
+  }, function(error, tokenResponse) {
+    if (error != null) {
+      prettyPrintResponse(error);
+      return response.json({
+        error: error,
+      });
+    }
+    ITEM_ADD_TOKEN = tokenResponse.add_token;
+    prettyPrintResponse(tokenResponse);
+    response.json({
+      add_token: ITEM_ADD_TOKEN,
+      expiration: tokenResponse.expiration,
+      error: null,
+    });
+  });
+});
+
+// Exchange token flow - exchange a Link public_token for
```

Comment:
We don't use any of the functions below this, let's remove them so developers have less to look at.